misc/synchronization.md

```diff
@@ -72,16 +72,22 @@ message to tiles. Therefore they will contain a generic exception.
 
 ## Matrix Views
 
-Matrix Views (`dlaf::matrix::MatrixView`) are objects which allow to schedule a part of the DAG of an
-algorithm in a different task.
+A matrix view can be can be used to create "sub-DAG" in the global DAG of a matrix.
+It gives the chance to take control of a particular asynchronous step of the matrix,
+allowing to concurrently schedule the work that will bbe executed after the work
+scheduled by the matrix view has been executed and the matrix view is notified that
+all the tasks has been scheduled (`done()`, `doneWrite()`, destructor).
```

Comment:
IMHO this sentence is quite long, you could evaluate to use two bullet points to give more structure to your thought:

>[...] allowing to concurrently schedule the work that will be executed:
>- after the work scheduled by the `MatrixView` has been executed, and
>- the `Matrix` ~view~ is notified that all the tasks has been scheduled (`done()`, `doneWrite()`, destructor).

Moreover, it is not so clear to me what you mean with the "2nd point". I think you are trying to say something like "the `MatrixView` will not schedule any additional task". In case, I would try to make it more clear.

Reply:
What I mean is that the work scheduled by the matrix will be executed after the work of the view only when the matrix view goes out of scope or done/doneWrite is called. Otherwise other work can be still scheduled by the view.